### DISCUSSION TITLE
Custom Types in Alembic migrations

### DISCUSSION THREAD

**Discussion by notypecheck** [sqlalchemy/sqlalchemy#8453] [Usage Questions] [answered]
Hello, I had to create a custom type to represent postgres `regconfig` type:
```py
class RegConfig(UserDefinedType):
    cache_ok = True

    def get_col_spec(self, **kw: Any) -> str:
        return "regconfig"

    def bind_processor(self, dialect: Dialect) -> Any:
        def process(value: str) -> str:
            return value

        return process

    def result_processor(self, dialect: Dialect, coltype: Any) -> Any:
        def process(value: str) -> str:
            return value

        return process
 ```
 I have a couple of questions:
 1. Does sqlalchemy has a built-in type for `regconfig`?
 2. Is there a "better" way of creating it to use in my model definition?
 3. A question regarding alembic - Is it possible for alembic to **not** reference that type in my code but either copy it or use a text (e.g. `"regconfig"` instead)? I don't want it to be dependent on my code base:
 ```py
from db.types import RegConfig  # The import in question

def upgrade():
    op.create_table(
        "...",
        sa.Column("column_name", RegConfig(), nullable=False),
        ...,
    )
```

**Comment by zzzeek** [comment, marked as answer]
we have "REGCLASS" but not "REGCONFIG".


your type does not need the bind_processor() type_processor() methods since they do nothing here.  so it only needs to be:

```py
class RegConfig(UserDefinedType):
    cache_ok = True

    def get_col_spec(self, **kw: Any) -> str:
        return "regconfig"
```


you should absolutely just copy the type into the migration files you need if you don't want to import it, it's a trivial stub.

**Reply by notypecheck** [reply]
Thanks, cutting size of this class helps, and yes, copying it into migration file is quite trivial.
